unit-tests/shared/src/test/scala/org/scalanative/testsuite/javalib/nio/channels/DatagramChannelTest.scala

```diff
@@ -0,0 +1,139 @@
+package org.scalanative.testsuite.javalib.nio.channels
+
```

Comment:
Later:  

I have just  read the `DatagramSocketTest` code
I think it existed before this new file and may have influence it.
This file looks to be more complete and is definitely more readable and understandable. This file is also, I believe, underlying transport agnostic and not IPv4 only.

An advancement of the art.